javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -951,8 +951,13 @@ protected void generateProtocolLongRunningBeginAsync(ClientMethod clientMethod,
     }
 
     private String getPollingStrategy(ClientMethod clientMethod, String contextParam) {
+        String endpoint = clientMethod.getProxyMethod().getParameters().stream()
+                .filter(p -> p.getRequestParameterLocation() == RequestParameterLocation.URI)
+                .findFirst().map(p -> p.getParameterReference()).orElse(null);
```

Comment:
See code here https://github.com/Azure/autorest.java/blob/main/javagen/src/main/java/com/azure/autorest/template/ServiceAsyncClientTemplate.java#L157

We don't want any host or URI, we want `endpoint`.